src/parse/common.rs

```diff
@@ -0,0 +1,74 @@
+use std::slice::IterMut;
+
+use codemap::Spanned;
+
+use crate::{value::Value, Token};
+
+pub(crate) struct NeverEmptyVec<T> {
+    first: T,
+    rest: Vec<T>,
+}
+
+impl<T> NeverEmptyVec<T> {
+    pub const fn new(first: T) -> Self {
+        Self {
+            first,
+            rest: Vec::new(),
+        }
+    }
+
+    pub fn last(&self) -> &T {
+        match self.rest.last() {
+            Some(v) => v,
+            None => &self.first,
+        }
```

Comment:
```suggestion
        self.rest.last().unwrap_or(&self.first)
```
Not tested but probably works, same goes for the one below.

Reply:
Thanks! Resolved in 822b9a35fefb8726814abda72d6b9f693b9f5352